src/timers/CTimers.cpp

```diff
@@ -0,0 +1,152 @@
+/// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2023 The MMapper Authors
+
+#include "CTimers.h"
+
+#include <QMutexLocker>
+#include <QSettings>
+#include <QTimer>
+
+CTimers::CTimers(QObject *parent)
+    : QObject(parent)
+{
+    m_nextId = 1;
+}
+
+CTimers::~CTimers() {}
+
+void CTimers::addTimer(QByteArray name, QByteArray desc)
+{
+    QMutexLocker locker(&m_lock);
+
+    TTimer timer;
+
+    timer.name = name;
+    timer.desc = desc;
+    timer.timer.start();
+    timer.id = m_nextId++;
+
+    m_timers.push_back(timer);
+}
+
+bool CTimers::removeCountdown(QByteArray name)
+{
+    QMutexLocker locker(&m_lock);
+
+    for (std::list<TTimer>::iterator it = m_countdowns.begin(); it != m_countdowns.end(); ++it) {
+        if (it->name == name) {
+            m_countdowns.erase(it);
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool CTimers::removeTimer(QByteArray name)
+{
+    QMutexLocker locker(&m_lock);
+
+    for (std::list<TTimer>::iterator it = m_timers.begin(); it != m_timers.end(); ++it) {
+        if (it->name == name) {
+            m_timers.erase(it);
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void CTimers::addCountdown(QByteArray name, QByteArray desc, int time)
+{
+    QMutexLocker locker(&m_lock);
+
+    TTimer countdown;
+    countdown.name = name;
+    countdown.desc = desc;
+    countdown.timer.start();
+    countdown.id = m_nextId++;
+    countdown.duration = time;
+
+    // append the shot-down event pending
+    QTimer::singleShot(time, this, SLOT(finishCountdownTimer()));
+
+    m_countdowns.push_back(countdown);
+}
+
+void CTimers::finishCountdownTimer()
+{
+    QMutexLocker locker(&m_lock);
+
+    int deleted = 0;
```

Comment:
Does it matter how many we delete? Can this be a boolean?

Reply:
Probably I'd better opt for writing out how many items were deleted. timers may have the same "name", so multiples may be deleted.

Reply:
Yeah, replaced with a boolean. 
Int wasn't necessary at all.